File: greedy/min_replacements/min_replacements.py

def count_min_operations(nums: list[int]) -> int:  # LeetCode Q.2366.
    min_operations = 0
    total_nums = len(nums)
    idx = -2  # Backward idx.
    while idx >= -total_nums:
        if nums[idx] > nums[idx + 1]:  # Maximize the division by minimizing pieces to break into.
            pieces = nums[idx] // nums[idx + 1]
            if nums[idx] % nums[idx + 1] > 0:  # Have to add another piece.
                pieces += 1

            nums[idx] //= pieces
            min_operations += pieces - 1  # Conduct (pieces - 1) operations.

        idx -= 1

    return min_operations

File: greedy/min_replacements/min_replacements.cpp
#include <vector>
using namespace std;

long long countMinOperations(vector<int> &nums)
{ // LeetCode Q.2366.
    long long minOperations = 0;

    int idx = nums.size() - 2;
    while (idx >= 0)
    {
        if (nums[idx] > nums[idx + 1])
        { // Maximize the division by minimizing pieces to break into.
            int pieces = nums[idx] / nums[idx + 1];

            if (nums[idx] % nums[idx + 1] > 0) // Have to add another piece.
                pieces += 1;

            nums[idx] /= pieces;
            minOperations += pieces - 1; // Conduct (pieces - 1) operations.
        }

        idx--;
    }
    return minOperations;
}
